api/nodes/utils.py

```diff
@@ -28,7 +28,10 @@ def get_file_object(node, path, provider, request):
     if not node.get_addon(provider) or not node.get_addon(provider).configured:
         raise NotFound('The {} provider is not configured for this project.'.format(provider))
 
-    url = waterbutler_api_url_for(node._id, provider, path, _internal=True, meta=True)
+    kwargs = {}
+    if 'view_only' in request.GET:
+        kwargs['view_only'] = request.query_params.get('view_only', default=None)
+    url = waterbutler_api_url_for(node._id, provider, path, _internal=True, meta=True, **kwargs)
```

Comment:
I lean towards being a little more explicit here and replacing this whole block with: 

```py
view_only = request.query_params.get('view_only', default=None)
url = waterbutler_api_url_for(node._id, provider, path, _internal=True, meta=True, view_only=view_only **kwargs)
```

That should do, essentially, the same thing, I think. You shouldn't need to check to se if `view_only` is in the request while also setting a default of None for getting the query parameter value.

That being said, if this is how we do it everywhere else, and or my way doesn't work, then your way is delightful and we'll go with that.